client/src/main/scala/com.gu.contentapi.client/utils/CapiModelEnrichment.scala

```diff
@@ -49,14 +56,144 @@ object CapiModelEnrichment {
+        "news/series/nauru-files",
+        "us-news/series/counted-us-police-killings",
+        "australia-news/series/healthcare-in-detention",
+        "society/series/this-is-the-nhs"
+      )
+
+      def isPillar(pillar: String): ContentFilter = content => content.pillarName.contains(pillar)
+
+      val isSpecialReport: ContentFilter = content => content.tags.exists(t => specialReportTags(t.id))
+      val isOpinion: ContentFilter = content => (isCommentDesign(content) && isPillar("News")(content)) ||
+        isPillar("Opinion")(content)
+      val isCulture: ContentFilter = content => isPillar("Arts")(content) || isPillar("Books")(content)
+
+      val predicates: List[(ContentFilter, Theme)] = List(
+        isOpinion -> OpinionPillar,
+        isPillar("Sport") -> SportPillar,
+        isCulture -> CulturePillar,
+        isPillar("Lifestyle") -> LifestylePillar,
+        isSpecialReport -> SpecialReportTheme,
+        tagExistsWithId("tone/advertisement-features") -> Labs,
+      )
+
+      val result = getFromPredicate(content, predicates)
+      result.getOrElse(defaultTheme)
+    }
+
+    def display: Display = {
+
+      val defaultDisplay = StandardDisplay
+
+      def hasShowcaseImage: ContentFilter = content => {
+        val hasShowcaseImage = for {
+          blocks <- content.blocks
+          main <- blocks.main
+          mainMedia = main.elements.head
+          imageTypeData <- mainMedia.imageTypeData
+          imageRole <- imageTypeData.role
+        } yield {
+          imageRole == "showcase"
+        }
+        hasShowcaseImage.getOrElse(false)
+      }
+
+      def hasShowcaseEmbed: ContentFilter = content => {
+
+        def isMainEmbed(elem: Element): Boolean = elem.relation == "main" && elem.`type` == ElementType.Embed
+
+        def hasShowcaseAsset(assets: scala.collection.Seq[Asset]): Boolean = {
+          val isShowcaseAsset = for {
+            embedAsset <- assets.find(asset => asset.`type` == AssetType.Embed)
+            typeData <- embedAsset.typeData
+            role <- typeData.role
+          } yield {
+            role == "showcase"
+          }
+          isShowcaseAsset.getOrElse(false)
+        }
+
+        val hasShowcaseEmbed = for {
+          elements <- content.elements
+          mainEmbed <- elements.find(isMainEmbed)
+        } yield {
+          hasShowcaseAsset(mainEmbed.assets)
+        }
+
+        hasShowcaseEmbed.getOrElse(false)
+      }
+
+      val isShowcase: ContentFilter = content => displayHintExistsWithName("column")(content) ||
+        displayHintExistsWithName("showcase")(content) ||
+        hasShowcaseImage(content) ||
+        hasShowcaseEmbed(content)
+
+      val isNumberedList: ContentFilter = displayHintExistsWithName("numberedList")
+
+      val predicates: List[(ContentFilter, Display)] = List(
+        isImmersive -> ImmersiveDisplay,
+        isShowcase -> ShowcaseDisplay,
+        isNumberedList -> NumberedListDisplay
```

Comment:
A note here that we need to update https://github.com/guardian/types/blob/main/src/format.ts#L37-L43 to remove `Column` Display.